Integrations/integration-HPE_Service_Manager.yml

```diff
@@ -0,0 +1,489 @@
+configuration:
+- display: Server URL (e.g. https://192.168.0.1:13080)
+  name: url
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Username
+  name: username
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: "false"
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: "false"
+  type: 8
+  required: false
+script:
+  script: |
+    var url = params.url + '/SM/9/rest';
+
+    // handle '/' at the end of the url
+    if (url[url.length - 1] === '/') {
+        url = url.substring(0, url.length - 1);
+    }
+
+    // var token = 'U3lzdGVtLkFkbWluOg==';
+    var token = Base64.encode(params.username.identifier + ':' + params.username.password);
+    // throw Base64.encode(params.username.identifier + ':' + params.username.password);
+
+    var result;
+    // The command input arg holds the command sent from the user.
+    switch (command) {
+        // This is the call made when pressing the integration test button.
+        case 'test-module':
+            result = 'ok';
+            break;
+        case 'hpsm-create-incident':
+            result = createIncident();
+            break;
+        case 'hpsm-list-incidents':
+            result = listIncidents();
+            break;
+        case 'hpsm-get-incident-by-id':
+            result = getIncidentById();
+            break;
+        case 'hpsm-list-devices':
+            result = listDevices();
+            break;
+        case 'hpsm-get-device':
+            result = getDeviceById();
+            break;
+        default:
+            // You can use args[argName] or args.argName to get a specific arg. args are strings.
+            // You can use params[paramName] or params.paramName to get a specific params.
+            // Params are of the type given in the integration page creation.
+    }
+    return result;
+
+    function createIncident() {
```

Comment:
does it work when functions defined at the end of the file

Reply:
yes it is. But when you define function like `function func_name()` and not `var func_name = function()`